pyrobosim/pyrobosim/core/room.py

```diff
@@ -126,10 +127,9 @@ def is_collision_free(self, pose):
         :rtype: bool
         """
         if isinstance(pose, Pose):
-            p = Point(pose.x, pose.y)
+            return intersects_xy(self.internal_collision_polygon, pose.x, pose.y)
         else:
-            p = Point(pose[0], pose[1])
-        return self.internal_collision_polygon.intersects(p)
+            return intersects_xy(self.internal_collision_polygon, pose[0], pose[1])
```

Comment:
This seems like a good improvement! 2 comments here:

First, can we unify the code so we only call the function once? Something like:

```suggestion
        if isinstance(pose, Pose):
            x = pose.x
            y = pose.y
        else:
            x = pose[0]
            y = pose[1]
        return intersects_xy(self.internal_collision_polygon, x, y)
```

Secondly, there are other instances of `Polygon.intersects()` being used beyond here. My search yields:

* `Hallway.is_collision_free()`
* `Location.is_inside()`
* `ObjectSpawn.is_inside()`